### DISCUSSION TITLE
Looking for an Offline-First solution for Mobile (iOS)

### DISCUSSION THREAD

**Discussion by Imaginativeone** [quasarframework/quasar#11790] [CLI - Capacitor / Cordova modes]
Do I need to save PouchDB data to SQLite? Please help.

**Comment by MilosPaunovic** [comment]
Hello, please try to explain your problem a bit more so we could see how it is Quasar related.

**Reply by Imaginativeone** [reply]
My problem has 2 parts:

**Current Progress**
- In my app, I'm using a fork of this component: https://github.com/ParadeTo/vue-tree-list, which features a recursive tree of data.
  - I was almost done with this app, using Ionic. However, I ran into this problem:
    - When I typed this in: **ionic cordova plugin add cordova-plugin-sqlite-2**
    - I got this response: **[ERROR] Ionic doesn't support using Cordova with @ionic/vue projects.**
 That's when I looked into Quasar: https://www.udemy.com/course/quasarframework/learn/lecture/15169108#overview

**Database Needs**
- My data can be represented by a JSON object, which changes when an item is moved from one tree location to another
- I want to save the user's JSON object, every time they make a change.
- I want to distribute an individual's changes among other users, using CouchDB.
- The BIGGEST problem that I want to solve is that I want to have unlimited space for my offline database.
- I don't want to deal with IndexedDB, WebSQL, etc., or anything else where the OS will wipe my offline/local data.

**Reply by MilosPaunovic** [reply]
That's all right, but I still can't figure out what is your Quasar related question...

**Comment by MatthewSH** [comment]
I'm going to write a separate response to answer things you mentioned [here](https://github.com/quasarframework/quasar/discussions/11790#discussioncomment-1875366).

Basically, what you're asking for isn't a Quasar specific thing. PWAs are offline capable, and Quasar supports PWA as a mode...so Quasar has support for being offline.

Your question is about saving offline data, how you do that (without being limited by size with session storage) is through IndexedDB.

> I don't want to deal with IndexedDB, WebSQL, etc., or anything else where the OS will wipe my offline/local data.

but you're title says:

> Looking for an Offline-First solution for Mobile (iOS) 

It's hard to figure out what you want when your entire thread is about offline functionality but the way to do that you don't want to use because can be wiped by the OS. If you're scared about saving it locally because it won't persist the data then just save it when it's online and if it's offline or the server is down then just alert them, save locally, and sync it back to server when it's back online. This is all functionality that's available with service workers and PWAs.
